### DISCUSSION TITLE
增加一个辅助方法来调用蓝图方法/事件

### DISCUSSION THREAD

**Discussion by xuyanghuang-tencent** [Tencent/UnLua#411] [Ideas]
当我们在蓝图里定义了一些函数或者事件，命名并不符合程序的套路，比如`Is any ally alive?`或`An enemy has been slain`。

这些命名在蓝图里都是合法的，但在Lua中如果需要访问则需要：

``` lua
-- 调用这个检测函数
if self["Is any ally alive?"](self) then
    -- 你不是一个人在战斗
end

-- 广播这个事件
self["An enemy has been slain"]:Broadcast()
```

是否引入一个辅助方法，来帮助代码风格上看起来更舒服一些：
``` lua
-- 在对象上占一个内置函数
self:Invoke("Do something awesome", ...)

-- 以内置库函数的形式
UnLua.Invoke(self, "Do something awesome", ...)
```

或者UnLua自身尽量保持简洁，上层的工作交给用户框架解决，欢迎讨论。

**Comment by CppCXY** [comment]
1. 如果提供了方法`Invoke`那我怎么简单的知道事件的名称和它具有的参数.
2. 如果采用`UnLua.Invoke`除了具有上面的问题,查找引用的话也很难知道这个事件在哪些地方广播，因为几乎所有lua文件都会调用这个公共函数。
